webrender/src/border.rs

```diff
@@ -345,18 +345,23 @@ impl BorderCornerClipSource {
             1.0 - 2.0 * outer_scale.y,
         );
 
+        // No point in pushing more clips as it will blow up the maximum amount of
+        // segments per primitive later down the road.
+        // See #2915 for a better fix.
+        let max_clip_count = self.max_clip_count.min(512);
```

Comment:
nit: should be computed from constants, e.g. `MAX_VERTEX_TEXTURE_WIDTH` divided by the blocks per segment